src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2686,27 +2689,65 @@ public class VerbArgumentCompleter : IArgumentCompleter
                 yield break;
             }
 
+            // Completion: Start-Process -FilePath <path> -Verb <wordToComplete>
             if (commandName.Equals(StartProcessCommandName, StringComparison.OrdinalIgnoreCase)
                 && fakeBoundParameters.Contains(FilePathParameterName))
             {
                 string filePath = fakeBoundParameters[FilePathParameterName].ToString();
 
+                // Complete file verbs if extension exists
                 if (Path.HasExtension(filePath))
                 {
-                    var verbPattern = WildcardPattern.Get(wordToComplete + "*", WildcardOptions.IgnoreCase);
+                    foreach (string verb in CompleteFileVerbs(filePath, wordToComplete))
+                    {
+                        yield return new CompletionResult(verb);
+                    }
+
+                    yield break;
+                }
+
+                // Otherwise check if command is an Application to resolve executable full path with extension
+                // e.g if powershell was given, resolve to powershell.exe to get verbs
+                using (var ps = System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace))
+                {
+                    var commandInfo = new CmdletInfo(GetCommandCommandName, typeof(GetCommandCommand));
+
+                    ps.AddCommand(commandInfo);
+                    ps.AddParameter(CommandParameterName, filePath);
 
-                    string[] verbs = new ProcessStartInfo(filePath).Verbs;
+                    Collection<CommandInfo> commands = ps.Invoke<CommandInfo>();
```

Comment:
Couldn't it be unacceptably slow?

Reply:
It didn't seem to have much of a performance issue. I think though any extra code for 😑 when extension is not given needs to be discussed if its worth adding. Tab completion can get very complicated quickly otherwise 😂